firmware/defmt-rtt/src/header.rs

```diff
@@ -0,0 +1,45 @@
+use core::sync::atomic::AtomicUsize;
+
+use crate::channel::Channel;
+
+// NOTE the `rtt-target` API is too permissive. It allows writing arbitrary data to any
+// channel (`set_print_channel` + `rprint*`) and that can corrupt defmt log frames.
+// So we declare the RTT control block here and make it impossible to use `rtt-target` together
+// with this crate.
+#[no_mangle]
+pub(crate) static mut _SEGGER_RTT: Header = Header {
+    id: *b"SEGGER RTT\0\0\0\0\0\0",
+    max_up_channels: 1,
+    max_down_channels: 0,
+    up_channel: Channel {
+        name: NAME as *const _ as *const u8,
+        buffer: unsafe { &mut BUFFER as *mut _ as *mut u8 },
+        size: SIZE,
+        write: AtomicUsize::new(0),
+        read: AtomicUsize::new(0),
+        flags: AtomicUsize::new(NOBLOCK_TRIM),
+    },
+};
```

Comment:
Is it problematic that the Header gets constructed outside of the `fn handle`? Will it get constructed "too early"?